docs/management/connectors/action-types/opsgenie.asciidoc

```diff
@@ -0,0 +1,173 @@
+
+The Opsgenie connector uses the https://docs.opsgenie.com/docs/alert-api[Opsgenie Alert API].
+
+[float]
+[[opsgenie-connector-configuration]]
+==== Connector configuration
+
+Opsgenie connectors have the following configuration properties.
+
+Name::      The name of the connector. The name is used to identify a  connector in the management UI connector listing, or in the connector list when configuring an action.
+URL::     The Opsgenie URL: https://api.opsgenie.com. For the EU instance of Opsgenie use: https://api.eu.opsgenie.com. If you are using the <<action-settings, `xpack.actions.allowedHosts`>> setting, make sure the hostname is added to the allowed hosts.
+API Key::   The Opsgenie API authentication key for HTTP Basic authentication. https://support.atlassian.com/opsgenie/docs/create-a-default-api-integration/[How to generate an Opsgenie API key].
+
+[float]
+[[opgenie-connector-networking-configuration]]
+==== Connector networking configuration
+
+Use the <<action-settings, Action configuration settings>> to customize connector networking configurations, such as proxies, certificates, or TLS settings. You can set configurations that apply to all your connectors or use `xpack.actions.customHostSettings` to set per-host configurations.
+
+[float]
+[[Preconfigured-opsgenie-configuration]]
+==== Preconfigured connector type
+
+[source,text]
+--
+ my-opsgenie:
+   name: preconfigured-opsgenie-connector-type
+   actionTypeId: .opsgenie
+   config:
+    apiUrl: https://api.opsgenie.com
+   secrets:
+     apiKey: apikey
+--
+
+Config defines information for the connector type.
+
+`apiUrl`:: A string that corresponds to *URL*
+
+Secrets defines sensitive information for the connector type.
+
+`apiKey`:: A string that corresponds to *API Key*.
+
+[float]
+[[define-opsgenie-ui]]
+==== Define connector in Stack Management
+
+Define Opsgenie connector properties.
+
+[role="screenshot"]
+image::management/connectors/images/opsgenie-connector.png[Opsgenie connector]
+
+Test Opsgenie action parameters.
+
+[role="screenshot"]
+image::management/connectors/images/opsgenie-params-test.png[Opsgenie params test]
+
+[float]
+[[opsgenie-action-configuration]]
+==== Action configuration
+
+The Opsgenie connector supports two types of actions: Creating and Closing alerts. The properties supported for each action are different because Opsgenie defines different properties for each operation.
+
+When testing the Opsgenie connector, choose the appropriate action from the selector. Switching the action will show a different form allowing the specific properties for that type of action to be configured.
+
+Action::    Whether to create or close an alert within Opsgenie. When using the Opsgenie connector for a rule action, the Action selector is hidden and automatically set to the appropriate value. If the action is run when a rule fires, it will create an alert within Opsgenie. If the action is run when a rule recovers, it will close an alert within Opsgenie.
```

Comment:
Hm, am I understanding correctly that an alert automatically closes when the rule that created it no longer detects the conditions that it (the rule) monitors for? If my understanding is right, does the following work?

```suggestion
Action::    Whether to create or close an alert within Opsgenie. The Action field is automatically set while you configure the connector for a rule. If the rule's parameters are met, an alert is created within Opsgenie. The alert is automatically closed once the rule no longer detects the conditions that generated the alert.
```

@lcawl separate question: do we need to create a shared attribute for Opsgenie?

Reply:
Since "Opsgenie" is a single term with no confusing capitalization etc, IMO we don't need a shared attribute. If there's a common acronym people use for this product or the product name might change imminently, then that would be a good reason.

Reply:
Here's an alternative version of the text to help simplify the terminology (and align with the way we describe recovery actions in https://www.elastic.co/guide/en/kibana/master/create-and-manage-rules.html):
```suggestion
Action:: Select *Create Alert* to configure the actions that occur when a rule's conditions are met. Select *Close Alert* to define the recovery actions that occur when the rule conditions are no longer met.
```